xarray/core/variable.py

```diff
@@ -412,15 +473,15 @@ def __setitem__(self, key, value):
 
         See __getitem__ for more details.
         """
-        key = self._item_key_to_tuple(key)
+        dims, index_tuple = self._broadcast_indexes(key)
         if isinstance(self._data, dask_array_type):
             raise TypeError("this variable's data is stored in a dask array, "
```

Comment:
This is actually no longer true -- `dask.array` supports assignment in recent versions.